graph/graph.go

```diff
@@ -141,14 +142,92 @@ func (graph *Graph) restore() error {
 		}
 	}
 
-	baseIds, err := graph.restoreBaseImages()
+	graph.idIndex = truncindex.NewTruncIndex(ids)
+	logrus.Debugf("Restored %d elements", len(ids))
+	return nil
+}
+
+func (graph *Graph) RestoreCustomImages(ts *TagStore) error {
```

Comment:
This was the function that we spoke about moving into the special interface implemented on the Windows graphdriver.  Unfortunately, because I need to be able to pass in the TagStore and make calls into methods on the graph object, I couldn't put it in our graphdriver because that would introduce a circular relationship (since graph imports graphdriver, which imports graphdriver/windows, graphdriver/windows can't import graph, which defines both Graph and TagStore).  Any thoughts on how we could hide this code in the graphdriver?  Short of copying a bunch of the graph and tag store implementation to allow the driver to read from and write to their folders, I couldn't come up with anything...